src/skan/csr.py

```diff
@@ -1092,3 +1094,121 @@ def make_degree_image(skeleton_image):
+        Pixel, not real, coordinates of point on skeleton keeping which as
+        center the concentric shells are computed. If None, pixel nearest to
+        geodesic center of skeleton is chosen, by default None
+    step_size : float or None, optional
+        Spacing between intermediate shells. If None, `num_shells` is used. It
+        takes precendence over `num_shells`, by default None
+    num_shells : int, optional
+        Number of concentric shells, by default None
+    start_radius : float, optional
+        The real world radius of the smallest shell, i.e., the first distance
+        to be sampled, by default 0
+    end_radius : float or None, optional
+        The real world radius of the largest (last) shell. If None, it is
+        automatically calculated as largest possible radius, by default None
+
+    Returns
+    -------
+    array
+        Radii in real world units for concentric shells used for analysis.
+    array
+        Number of intersections for corresponding shell radii.
+    """
+    def _path_distances(skeleton, center_point, path_id):
+        """Compute real world distances of specific skeleton path coordinates
+        from the center point.
+
+        Parameters
+        ----------
+        skeleton : skan.csr.Skeleton
+            A Skeleton object.
+        center_point : array
+            Real world coordinates of center.
+        path_id : int
+            Path ID of path to be traversed.
+
+        Returns
+        -------
+        ndarray
+            Distance from each pixel in the path to the central pixel in real
+            world units.
+        """
+        path = skeleton.path_coordinates(path_id)
+        path_scaled = path * skeleton.spacing
+        distances = np.ravel(distance_matrix(path_scaled, [center_point]))
+        return distances
+
+    # Compute vector magnitude (length)
+    _magnitude = lambda coords: np.sqrt(np.sum(coords ** 2))
+
+    imskeleton = skeleton.skeleton_image
+    skel_shape = np.asarray(imskeleton.shape)
+
+    if soma is None:
+        soma, _ = central_pixel(skeleton.graph, shape=skel_shape,
+                                nodes=np.transpose(skeleton.coordinates))
+
+    soma = np.asarray(soma)
+
+    # if soma doesn't lie on skeleton, find nearest point on skeleton
+    if not imskeleton[tuple(soma)]:
+        soma = min(skeleton.coordinates,
+                   key=lambda x: _magnitude(soma - x))
+
+    scaled_soma = soma * skeleton.spacing
+
+    if end_radius is None:
+        end_radius = _magnitude(skel_shape * skeleton.spacing) / 2
+
+    if num_shells is None and type(step_size) in (int, float):
+        shell_radii = np.arange(start_radius,
+                                end_radius + step_size,
+                                step_size)
+    else:
+        if step_size is None and num_shells is None:
+            # TODO: set num_shells to # pixels away from farthest
+            # vertex of image
```

Comment:
I think a better default is half or even a quarter of the number of pixels. Single pixel measures are subject to noise.

Reply:
For this I'm thinking of using the same approach I proposed for `end_radius` auto-initialization. Just that the euclidian distances would be calculated in pixel units and I'll take the np.floor of the max